src/node_contextify.cc

```diff
@@ -270,6 +271,44 @@ void ContextifyContext::IsContext(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void ContextifyContext::WrapFunction(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Local<Context> context = env->context();
+
+  CHECK(args[0]->IsString());
+  Local<String> source = args[0].As<String>();
+  ScriptCompiler::Source script_source(source);
+
+  Local<String> params[] = {
+    env->module_parameter_exports(),
+    env->module_parameter_require(),
+    env->module_parameter_module(),
+    env->module_parameter_filename(),
+    env->module_parameter_dirname()
+  };
+
+  TryCatch try_catch(env->isolate());
+  Environment::ShouldNotAbortOnUncaughtScope no_abort_scope(env);
+  Context::Scope scope(context);
+
+  MaybeLocal<Function> maybe_fun = ScriptCompiler::CompileFunctionInContext(
+      context, &script_source, 5, params, 0, nullptr);
+
+  if (maybe_fun.IsEmpty()) {
+    no_abort_scope.Close();
+    try_catch.ReThrow();
+    return;
+  }
+
+  Local<Function> fun = maybe_fun.ToLocalChecked();
+  CHECK(!fun.IsEmpty());
+  CHECK(!try_catch.HasCaught());
+
+  Local<String> result = fun->ToString(context).ToLocalChecked();
```

Comment:
This is precisely what we do not want to do. Do not turn the function into a string. We want the function itself.